src/games/strategy/triplea/attachments/TerritoryAttachment.java

```diff
@@ -138,15 +138,7 @@ public static Territory getFirstOwnedCapitalOrFirstUnownedCapital(final PlayerID
    * Convenience method. Can return null.
    */
   public static TerritoryAttachment get(final Territory t) {
-    return get(t, false);
-  }
-
-  public static TerritoryAttachment get(final Territory t, final boolean allowNull) {
-    final TerritoryAttachment rVal = (TerritoryAttachment) t.getAttachment(Constants.TERRITORY_ATTACHMENT_NAME);
-    if (!allowNull && rVal == null && !t.isWater()) {
-      throw new IllegalStateException("No territory attachment for:" + t.getName());
-    }
-    return rVal;
+    return (TerritoryAttachment) t.getAttachment(Constants.TERRITORY_ATTACHMENT_NAME);
```

Comment:
The test set up had a failure here. Looking at it, we are being overly restrictive. All usages of `get(final Territory t)` I could spot check for null return values. Though consider the case where we forgot a NPE check on return value and so we actually relied on the IllegalStateException. In that case we should still get a NPE since we're returning a null value, and the code will still error out. It's the same error behavior.

To be a bit more clear about what happened here though, we are putting the burden on the caller to require a non-null response. Note that we are actually in effect defaulting to 'true'. We are removing the condition controlled by `allowNull` altogether, effectively setting it to true (note the `!` that negates it, and when `allowNull` is true then we will never execute the if statement)